src/main/java/org/apache/commons/io/FileUtils.java

```diff
@@ -377,8 +378,12 @@ public static boolean contentEquals(final File file1, final File file2) throws I
             return true;
         }
 
-        checkFileExists(file1, "file1");
-        checkFileExists(file2, "file2");
+        if (!file1.isFile()) {
+            throw new IllegalArgumentException("Parameter '" + "file1" + "' is not a file: " + file1);
```

Comment:
Hello @elharo 
This must be a typo: `"Parameter '" + "file1" + "' is not a file: "`
There is now the same (weird) pattern x3
```
        if (!file2.isFile()) {
            throw new IllegalArgumentException("Parameter '" + "file2" + "' is not a file: " + file2);
        }
```
This was the point of the `requireX()` method, so this should be fixed and refactored. But then, this looks the same as the existing `requireFile()` method... what am I missing here?

Reply:
checkFileExists also includes a check that the file exists and throws a FileNotFoundException if it does not. However, at this point in the method it is already known that the files exist, so we don't want to check that again.

Reply:
Hello @elharo 
The weird string construction is still bogus IMO and the pattern in triplicate should be refactored into a new private method.

Reply:
OK. I see what you're saying now. I'll combine the strings. I did this elsewhere. It's just an artifact of Eclipse inlining.